pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -308,3 +310,82 @@ func (s *AWSCloud) ensureLoadBalancerInstances(loadBalancerName string, lbInstan
 
 	return nil
 }
+
+func getLoadBalancerHostedZone(service *api.Service) string {
+	if zone, ok := service.Annotations[ServiceAnnotationLoadBalancerCnameZone]; ok {
+		return zone + "."
+	} else {
+		return ""
+	}
+}
+
+func (s *AWSCloud) getLoadBalancerCname(service *api.Service) string {
+	hostedZone := getLoadBalancerHostedZone(service)
+	if hostedZone == "" {
+		return ""
+	}
+
+	return fmt.Sprintf("%v-%v-%v.%v", service.Name, service.Namespace, s.cfg.Global.KubernetesClusterTag, hostedZone)
```

Comment:
Our main use case for this is to give a stable name to services for internal use.

We plan to use ingress for hostnames.